node/node.go

```diff
@@ -1002,30 +1002,14 @@ func (node *AlgorandFullNode) loadParticipationKeys() error {
 }
 
 func insertStateProofToRegistry(part account.PersistedParticipation, node *AlgorandFullNode) error {
-	// create a const value Number of records in one fetch
-	// insert into the registry X rounds of keys via account manager
-	// all in for loop until no more keys in db
 	partID := part.ID()
-	numKeys, err := part.Participation.StateProofSecrets.CountKeys(part.Store)
-	if err != nil {
-		return err
-	}
-	keys := make(map[uint64]account.StateProofSigner, numKeys)
-	for i := uint64(0); i < uint64(numKeys); i++ {
-		key, round, err := part.Participation.StateProofSecrets.FetchKey(i, part.Store)
-		if err != nil {
-			return err
-		}
-
-		keys[round] = account.StateProofSigner(*key)
-	}
-	err = node.accountManager.Registry().AppendKeys(partID, keys)
-	// kinda useless since AppendKeys returns nil exclusively and appends asynchronously
-	if err != nil {
-		return err
+	keys := part.StateProofSecrets.GetAllKeys()
```

Comment:
i'm not sure it needs to be a different function anymore, since this can only be used here and only if part was created by `RestoreParticipationWithSecrets`